packages/flask/src/UICarouselView/CarouselView/CarouselContainer/index.web.tsx

```diff
@@ -0,0 +1,189 @@
+    key: any;
+};
+
+const PageView: React.FC<PageProps> = React.memo(
+    ({
+        page,
+        index,
+        currentIndex,
+        width,
+        offset,
+        onPress,
+        shouldPageMoveOnPress,
+        key,
+    }: PageProps) => {
+        const PageComponent = page.props.component;
+
+        const animatedStyles = useAnimatedStyle(() => {
+            const opacity = interpolate(offset.value, [1, 0], [0, 1]);
+            const transform = [
+                {
+                    scale: interpolate(offset.value, [1, 0], [0.9, 1]),
+                },
+            ];
+            return { opacity, transform };
+        });
+
+        const focused = React.useMemo(() => index === currentIndex.current, [currentIndex]);
+
+        const pageWidthStyle = React.useMemo(() => {
+            return width ? { width } : focused ? { ...StyleSheet.absoluteFillObject } : null;
+        }, [width, focused]);
+
+        return (
+            <Pressable
+                disabled={!shouldPageMoveOnPress}
+                onPress={onPress}
+                key={key}
+                testID={page.props.testID}
+            >
+                <Animated.View style={[animatedStyles, pageWidthStyle]}>
+                    {width ? <PageComponent /> : null}
+                </Animated.View>
+            </Pressable>
+        );
+    },
+);
+
+type Props = UICarouselViewContainerProps & {
+    pages: React.ReactElement<UICarouselViewPageProps>[];
+};
+
+export const CarouselViewContainer: React.FC<Props> = ({
+    pages,
+    initialIndex = 0,
+    testID,
+    shouldPageMoveOnPress = true,
+    onPageIndexChange,
+}: Props) => {
+    const [layout, setLayout] = React.useState({ width: 0, height: 0 });
+
+    const carouselOffset = useSharedValue(0);
+    const itemOffset = useSharedValue(0);
+
+    const layoutRef = React.useRef(layout);
+    const onPageChangedRef = React.useRef(onPageIndexChange);
+    const currentIndexRef = React.useRef(0);
+    const maxWidthTranslate = React.useMemo(() => layout.width * pages.length, [layout, pages]);
+
+    const [isMoving, setIsMoving] = React.useState(false);
+
+    const containerAnimatedStyle = useAnimatedStyle(() => {
+        const transform = [
+            {
+                translateX: interpolate(
+                    carouselOffset.value,
+                    [-maxWidthTranslate, 0],
+                    [-maxWidthTranslate, 0],
+                ),
+            },
+        ];
+
+        return { transform };
+    });
+
+    const jumpToIndex = React.useCallback(
+        (index: number) => {
+            const offset = -index * layout.width;
+            carouselOffset.value = withTiming(offset, { duration });
+            itemOffset.value = withSequence(
+                withTiming(1, { duration: duration / 2 }),
+                withTiming(0, { duration }),
+            );
+            setIsMoving(false);
+            onPageChangedRef.current && onPageChangedRef.current(index);
```

Comment:
I actually didn't get what this optimization is about :) Can you please enlighten me a bit, what was the reasoning?

Reply:
@savelichalex In this case, saving to ref makes the callback more stable. I don't really understand what is the reason for errors without ref, because I did not delve into it.

Reply:
Oh, I guess it's because you didn't pass it to deps array for `useEffect`

Reply:
Please try to remove it, as for now it looks like an unpredictable hack :)